src/lib/modules/module/monitor.go

```diff
@@ -48,35 +54,35 @@ func (m *Monitor) Run() {
 		// canceling scheduler if already present.
 		cancelCurrentScheduler(cancel)
 		ctx, cancel = context.WithCancel(context.Background())
-		scheduler := scheduler.New(m.jobs)
-		go scheduler.Run(ctx)
+		schdr := scheduler.New(m.jobs)
+		go schdr.Run(ctx)
 	}
 }
 
-// Reload reloads the new config and signals reload channel.
-func (m *Monitor) Reload(conf *config.Config, errCh chan<- error) {
-	jobs := make(map[*job.JobInfo]chan<- struct{})
-	set := file.NewChainSet(0, time.Second*10)
-	set.Run()
+// Reload reloads the new config and signals reload channel. It expects the Run() to already be active.
+func (m *Monitor) Reload(conf *config.Config) error {
 	for i, api := range conf.APIs {
-		app, _ := set.NewChain(api.Name, api.Domain+api.Route, false)
+		_, exists := m.existingJobs[api.Name]
+		if exists {
+			// todo: deletion of jobs that are no longer existing
+			log.Info("component", "reload", "msg", "job already exists with name "+api.Name+". Skipping creation.")
+			continue
+		}
+		app, _ := m.chainSet.NewChain(api.Name+"_monitor", api.Domain+api.Route, false)
 
-		// creating the jobs.
 		exec, ch, err := job.NewJob("monitor", app, &api)
 		if err != nil {
-			errCh <- fmt.Errorf("error creating # %d job: %s", i, err)
-			continue
+			return fmt.Errorf("error creating # %d job: %s", i, err)
 		}
-		// launching the jobs.
-		go exec.Execute()
-		jobs[exec.Info()] = ch
+		go exec.Execute(m.errCh)
+		m.jobs[exec.Info()] = ch
+		m.existingJobs[api.Name] = struct{}{}
 	}
 	m.mux.Lock()
-	m.jobs = jobs
 	m.mux.Unlock()
```

Comment:
I guess since we are not using extra variables for the m.jobs update, we should remove the mux lock and unlock.